packages/context/src/watcher.ts

```diff
@@ -265,6 +255,70 @@ export const reactOnXmlFileChange = async (
   await getProject(documentPath);
 };
 
+/**
+ * React to an `.view.xml` or `.fragment.xml` file change.
+ *
+ * @param uri uri to an view file
+ * @param changeType change type
+ * @description
+ * a. for remove operation, it delete view file and controls ids entry
+ *
+ * b. for create operation, it add view file and controls ids entry
+ *
+ * c. trigger validator at last to revalidate
+ */
+export async function reactOnViewFileChange(
+  uri: string,
+  changeType: FileChangeType,
+  validator: () => Promise<void>
+): Promise<void> {
+  getLogger().debug("`reactOnViewFileChange` function called", {
+    xmlUri: uri,
+    changeType,
+  });
+
+  const documentPath = URI.parse(uri).fsPath;
+  if (!isXMLView(documentPath)) {
+    return;
+  }
+
+  const manifestPath = await findManifestPath(documentPath);
+  if (!manifestPath) {
+    return;
+  }
+  if (changeType === FileChangeType.Deleted) {
```

Comment:
Both blocks look exactly the same, just with different operation. We can just pass the `operation` based on the `changeType`.